src/Catalog/Icons/HttpResponseMessageProvider.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace NuGet.Services.Metadata.Catalog.Icons
+{
+    public class HttpResponseMessageProvider : IHttpResponseMessageProvider
```

Comment:
>IHttpResponseMessageProvider [](start = 47, length = 28)

Thoughts on naming these `IHttpClient` and `HttpClientWrapper`? A little shorter

Reply:
That would assume some sort of feature parity with `HttpClient`, which it absolutely does not have.

Reply:
Yes, we're cutting corners by not adding all `HttpClient` methods to `IHttpClient`, but does it matter? Is there a reason why one couldn't add missing `HttpClient` methods to `HttpResponseMessageProvider` in the future?

We use this pattern already for blob storage ([example](https://github.com/NuGet/NuGetGallery/blob/master/src/NuGetGallery.Core/Services/CloudBlobContainerWrapper.cs)).